File: ProblemSet/Week 2/Problem/Python/List.py
def main():
    # Create a list to store K-POP groups
    kpop = []

    # Ask the user to enter the number of K-POP groups to add to the list
    num_groups = int(input("Enter the number of K-POP groups to add to the list: "))

    # Ask the user to enter the K-POP groups to add to the list
    print("Enter the K-POP groups to add to the list:")
    for i in range(num_groups):
        group = input(f"Group {i+1}: ")
        kpop.append(group)

    # Print the list elements
    print("List KPOP:")
    for group in kpop:
        print(group, end=" ")
    print()

    # Ask the user if they want to remove a K-POP group from the list
    remove_group = input("Do you want to remove a K-POP group from the list? (yes/no): ")
    if remove_group.lower() == "yes":
        
        # Ask the user to enter the K-POP group to remove
        group_to_remove = input("Enter the K-POP group to remove: ")
        if group_to_remove in kpop:
            kpop.remove(group_to_remove)
            print("Updated list KPOP:")
            for group in kpop:
                print(group, end=" ")
            print()
        else:
            print("Group not found in the list.")
    else:
        print("No K-POP group removed from the list.")

    # Ask the user if they want to add another K-POP group to the list
    add_group = input("Do you want to add another K-POP group to the list? (yes/no): ")
    if add_group.lower() == "yes":
        
        # Ask the user to enter the new K-POP group
        new_group = input("Enter the new K-POP group: ")
        kpop.append(new_group)
        print("Updated list KPOP:")
        for group in kpop:
            print(group, end=" ")
        print()
    else:
        print("No new K-POP group added to the list.")

if __name__ == "__main__":
    main()

File: ProblemSet/Week 2/Problem/JavaScript/List.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let kpop = [];

/**
 * Asks the user to enter the number of K-POP groups to add to the list
 * and then asks for each group individually.
 */
rl.question('Enter the number of K-POP groups to add to the list: ', (numGroups) => {
  numGroups = parseInt(numGroups);
  let count = 0;
  
  /**
   * Recursive function to ask for each K-POP group individually.
   */
  const askGroup = () => {
    rl.question(`Group ${count + 1}: `, (group) => {
      kpop.push(group);
      count++;
      if (count < numGroups) {
        askGroup(); // Recursively ask for the next group
      } else {
        console.log('List KPOP:');
        console.log(kpop.join(' '));
        askRemove(); // Move on to ask about removing a group
      }
    });
  };
  askGroup(); // Start asking for groups
});

/**
 * Asks the user if they want to remove a K-POP group from the list.
 */
const askRemove = () => {
  rl.question('Do you want to remove a K-POP group from the list? (yes/no): ', (removeGroup) => {
    if (removeGroup.toLowerCase() === 'yes') {
      rl.question('Enter the K-POP group to remove: ', (groupToRemove) => {
        const index = kpop.indexOf(groupToRemove);
        if (index !== -1) {
          kpop.splice(index, 1);
          console.log('Updated list KPOP:');
          console.log(kpop.join(' '));
        } else {
          console.log('Group not found in the list.');
        }
        askAdd(); // Move on to ask about adding a new group
      });
    } else {
      console.log('No K-POP group removed from the list.');
      askAdd(); // Move on to ask about adding a new group
    }
  });
};

/**
 * Asks the user if they want to add another K-POP group to the list.
 */
const askAdd = () => {
  rl.question('Do you want to add another K-POP group to the list? (yes/no): ', (addGroup) => {
    if (addGroup.toLowerCase() === 'yes') {
      rl.question('Enter the new K-POP group: ', (newGroup) => {
        kpop.push(newGroup);
        console.log('Updated list KPOP:');
        console.log(kpop.join(' '));
        rl.close(); // Close the readline interface
      });
    } else {
      console.log('No new K-POP group added to the list.');
      rl.close(); // Close the readline interface
    }
  });
};
